src/backend/metal/src/window.rs

```diff
@@ -9,8 +9,11 @@ use hal::{self, format, image};
 use hal::{Backbuffer, SwapchainConfig};
 use hal::window::Extent2D;
 
-use cocoa::foundation::{NSRect};
 use core_graphics::geometry::{CGSize};
+#[cfg(target_os = "macos")]
+use cocoa::foundation::NSRect as OSRect;
```

Comment:
Could we just use `CGSize` for both or is this problematic?

Reply:
That would work.

According to [this](https://developer.apple.com/documentation/foundation/nsrect?language=objc) `NSRect` is typedef'd to `CGRect` on 64-bit platforms, and according to [this](https://support.apple.com/en-us/HT208436) metal only works on 64-bit OSX (also true for iOS).